enhancements/insights/pulling-pull-secret-from-ocm.md

```diff
@@ -0,0 +1,168 @@
+
+1. OCM provides an API that the cluster can call to check the current global `pull-secret`.
+2. User starts the migration process in the OCM by providing the `pull-secret` associated with the account to which the ownership will be transferred to.
+   * The details of whether the user provides the pull secret as a means of verification, or whether a request/approval workflow is implemented, is left to the OCM/UI flow. What is pertinent here is that there is an endpoint available to a cluster entity to fetch the new `pull-secret`.
+3. OCM validates the `pull-secret` for the new account and makes it available to the cluster for some time period i.e. the new API will now return the new `pull-secret` provided by the user.
+4. The Insights operator calls this API periodically, fetches the `pull-secret` and compares it with the current global `pull-secret` on the cluster.
+5. If there is a mismatch, the Insights operator applies the new `pull-secret`.
+   * Note: MCO updating the `pull-secret` on the nodes will not result in node reboots.
+6. The Insights operator itself starts using the new `pull-secret`
+7. Cluster comes up with the new `pull-secret` and starts reporting back to OCM into the new account.
+
+Note: This process will work only on the clusters where the feature was released as part of the Insights operator and will not be backported to older versions.
+
+### User Stories
+
+As an OCM user, I would like to transfer cluster ownership from my account to another user's account in one step.
+
+### API Extensions
+
+No API extensions.
+
+### Risks and Mitigations
+
+Risk: The OCM API is down or returning HTTP response code >= 500.
+
+Mitigation: The Insights Operator is marked as degraded in such case.
+
+Risk: The `pull-secret` does not work or is otherwise corrupted.
+
+Mitigation: We must refer to the OCP documentation, which describes the manual fix procedure. 
+
+
+## Design Details
+
+### Authorization
+
+The Insights Operator is able to pull the updated `pull-secret` content from the OCM API using the existing `cloud.openshift.com` token
+available in the `pull-secret` (in the `openshift-config` namespace).
+
+The Insights Operator must provide a cluster ID as an identifier of the cluster.
+
+### OCM API
+
+The OCM API provides an endpoint that will serve the complete `pull-secret` definition when the migration window is opened. 
+The migration window will be opened for certain period of time (e.g 5 days) so tha the Insights Operator can safely update the secret.
+
+Open question: What exactly will the API return outside of the given migration window?  
+
+### Monitoring
+
+There will be a new metric registered by the Insights Operator, which will count the number of responses (according the HTTP response status code) from the OCM API.
```

Comment:
And then what happens to this metric?  Is it covered by alerting?  Or exported via Telemetry/Insights?  Or...?

Reply:
I think most likely alerting....and then maybe an Insights rule